src/ops/discard.rs

```diff
@@ -10,20 +10,22 @@ impl OpTrait for Discard {
     fn get_action(&self, target: Option<&TargetData>) -> Option<Action> {
         let action = match target.cloned() {
             Some(TargetData::Branch(r)) => cmd_arg(discard_branch, r.into()),
-            Some(TargetData::File(f)) => Rc::new(move |state: &mut State, _term: &mut Term| {
-                let path = PathBuf::from_iter([
-                    state.repo.workdir().expect("No workdir").to_path_buf(),
-                    f.clone(),
-                ]);
-                std::fs::remove_file(path)?;
-                state.screen_mut().update()
-            }),
+            Some(TargetData::File(f)) => cmd_arg(clean_file_cmd, f.into()),
```

Comment:
Optional: I had a look at why there's TargetData::File and TargetData::Delta, and found some refactoring opportunity. It should be possible to remove TargetData::File completely removed from the code-base!

Reply:
There is one place where TargetData::File is created and it is in items_list. I tried to replace it with delta but got confused pretty quick. I think I need to look into that one more time later...